### PR TITLE
COMP: Remove support for VTK8

### LINKED ISSUE #5985: Streamline maintenance removing support for VTK8
## Is your feature request related to a problem? Please describe.

Considering that:
* there are no continuous integration testing the build of Slicer with option `Slicer_VTK_VERSION_MAJOR` set to `8`
* efforts are focused toward building against the latest version of VTK (see https://github.com/Slicer/Slicer/issues/5956)
* VTK8 is not maintained

the overall maintenance should be simplified by removing VTK8 support.

## Describe the solution you'd like

* [ ] Update [SuperBuild/External_VTK.cmake](https://github.com/Slicer/Slicer/blob/master/SuperBuild/External_VTK.cmake)
* [ ] Remove conditional branching based of `VTK_MAJOR_VERSION` in [C++ files](https://github.com/Slicer/Slicer/search?q=VTK_MAJOR_VERSION&type=code)
* [ ] Removing conditional branching based of `VTK_VERSION` in [CMake modules](https://github.com/Slicer/Slicer/search?l=CMake&q=VTK_VERSION&type=code)

## Describe alternatives you've considered

NA

## Additional context

NA

### PR DESCRIPTION
**Pull request by jamesobutler**
This closes #5985, a Slicer 5.1 milestone issue.

Build support for VTK8 or VTK9 was introduced in October 2020 in https://github.com/Slicer/Slicer/commit/c658b7b428a37d6f4e925664099ee30fd4ca408c when VTK9 support was added.

To streamline maintenance, VTK8 support is now being removed.

Note that other projects that might still be using VTK8 with an older Slicer version appear to actively be upgrading their Slicer version and therefore VTK version. See SlicerSalt upgrading to Slicer 5 in https://github.com/Kitware/SlicerSALT/pull/268.

I've confirmed with a fresh build that things build successfully on Windows, that packaging works and that all tests continue to pass.

### CODE REVIEW

**Review by pieper — APPROVED** [review]
Makes good sense to me to drop this.

**Comment by cpinter** [comment]
Agreed, thank you!

File: SuperBuild/External_VTK.cmake [hunk 1]
```diff
@@ -163,10 +138,7 @@ if((NOT DEFINED VTK_DIR OR NOT DEFINED VTK_SOURCE_DIR) AND NOT Slicer_USE_SYSTEM
     )
 
   set(_git_tag)
-  if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "8")
-    set(_git_tag "97904fdcc7e73446b3131f32eac9fc9781b23c2d") # slicer-v8.2.0-2018-10-02-74d9488523
-    set(vtk_egg_info_version "8.2.0")
-  elseif("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "9")
+  if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "9")
```
Comment (jcfr) [inline SuperBuild/External_VTK.cmake:141]: In the past, when removing support for older VTK, to ensure using git blame report the original motivation leading to the integration of specific code .... I kept the indent (with/without keeping the associated `if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "9")` like done here.

File: SuperBuild/External_SlicerExecutionModel.cmake [hunk 2]
```diff
@@ -49,17 +49,15 @@ if(NOT DEFINED SlicerExecutionModel_DIR AND NOT Slicer_USE_SYSTEM_${proj})
...
-      # Required by FindPython3 CMake module used by VTK
-      -DPython3_ROOT_DIR:PATH=${Python3_ROOT_DIR}
-      -DPython3_INCLUDE_DIR:PATH=${Python3_INCLUDE_DIR}
-      -DPython3_LIBRARY:FILEPATH=${Python3_LIBRARY}
-      -DPython3_LIBRARY_DEBUG:FILEPATH=${Python3_LIBRARY_DEBUG}
-      -DPython3_LIBRARY_RELEASE:FILEPATH=${Python3_LIBRARY_RELEASE}
-      -DPython3_EXECUTABLE:FILEPATH=${Python3_EXECUTABLE}
-      )
-  endif()
+  list(APPEND EXTERNAL_PROJECT_OPTIONAL_CMAKE_CACHE_ARGS
```
Comment (jcfr) [inline SuperBuild/External_SlicerExecutionModel.cmake:52]: Here, we should keep the block inside `if(Slicer_USE_PYTHONQT)`

File: SuperBuild/External_ParameterSerializer.cmake [hunk 3]
```diff
@@ -34,17 +34,15 @@ if(NOT DEFINED ${proj}_DIR AND NOT Slicer_USE_SYSTEM_${proj})
...
-      # Required by FindPython3 CMake module used by VTK
-      -DPython3_ROOT_DIR:PATH=${Python3_ROOT_DIR}
-      -DPython3_INCLUDE_DIR:PATH=${Python3_INCLUDE_DIR}
-      -DPython3_LIBRARY:FILEPATH=${Python3_LIBRARY}
-      -DPython3_LIBRARY_DEBUG:FILEPATH=${Python3_LIBRARY_DEBUG}
-      -DPython3_LIBRARY_RELEASE:FILEPATH=${Python3_LIBRARY_RELEASE}
-      -DPython3_EXECUTABLE:FILEPATH=${Python3_EXECUTABLE}
-      )
-  endif()
+  list(APPEND EXTERNAL_PROJECT_OPTIONAL_CMAKE_CACHE_ARGS
```
Comment (jcfr) [inline SuperBuild/External_ParameterSerializer.cmake:37]: Keep `if(Slicer_USE_PYTHONQT)`

File: SuperBuild/External_ITK.cmake [hunk 4]
```diff
@@ -57,17 +57,15 @@ if(NOT DEFINED ITK_DIR AND NOT Slicer_USE_SYSTEM_${proj})
...
-        # Required by FindPython3 CMake module used by VTK
-        -DPython3_ROOT_DIR:PATH=${Python3_ROOT_DIR}
-        -DPython3_INCLUDE_DIR:PATH=${Python3_INCLUDE_DIR}
-        -DPython3_LIBRARY:FILEPATH=${Python3_LIBRARY}
-        -DPython3_LIBRARY_DEBUG:FILEPATH=${Python3_LIBRARY_DEBUG}
-        -DPython3_LIBRARY_RELEASE:FILEPATH=${Python3_LIBRARY_RELEASE}
-        -DPython3_EXECUTABLE:FILEPATH=${Python3_EXECUTABLE}
-        )
-    endif()
+    list(APPEND EXTERNAL_PROJECT_OPTIONAL_CMAKE_CACHE_ARGS
```
[the remark this answers is not in the document]
Comment (jcfr) [reply SuperBuild/External_ITK.cmake:60]: > then a follow-up commit to fix the indent [...] `.git-blame-ignore-revs`

That would work very well.

[the remark this answers is not in the document]
Comment (jamesobutler) [reply SuperBuild/External_ITK.cmake:60]: Are you desiring the inconsistent indent followed by the fixing up the indent in the various files where there was the branching of the VTK version? and not just the ITK one where you commented here?
```
    # pythonpath
    if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "9")
      set(_library_install_subdir "bin")
      if(UNIX)
        set(_library_install_subdir "lib")
      endif()
    else()
      set(_library_install_subdir ${Slicer_INSTALL_BIN_DIR})
      if(UNIX)
        set(_library_install_subdir ${Slicer_INSTALL_LIB_DIR})
      endif()
```
to the following which maintains the indent
```
    # pythonpath
      set(_library_install_subdir "bin")
      if(UNIX)
        set(_library_install_subdir "lib")
      endif()
```
followed by fixing up the indent in another commit
```diff
    # pythonpath
-      set(_library_install_subdir "bin")
-      if(UNIX)
-        set(_library_install_subdir "lib")
-      endif()
+    set(_library_install_subdir "bin")
+    if(UNIX)
+      set(_library_install_subdir "lib")
+    endif()
```

**Review by lassoan — COMMENTED** [review]
Awesome, thank you, I just added two comments.

File: CMakeLists.txt [hunk 5]
```diff
@@ -552,13 +552,8 @@ message(STATUS "  Slicer_VTK_VERSION_MAJOR is ${Slicer_VTK_VERSION_MAJOR}")
...
 #-----------------------------------------------------------------------------
-if (Slicer_VTK_VERSION_MAJOR VERSION_LESS "9")
-  set(VTK_COMPONENT_PREFIX "vtk")
-  set(VTK_TARGET_PREFIX "vtk")
-else()
-  set(VTK_COMPONENT_PREFIX "")
-  set(VTK_TARGET_PREFIX "VTK::")
-endif()
+set(VTK_COMPONENT_PREFIX "")
+set(VTK_TARGET_PREFIX "VTK::")
```
[the remark this answers is not in the document]
Comment (jamesobutler) [reply CMakeLists.txt]: I've pushed an additional commit here for this replacement. Note that I haven't done a sanity build test yet with these changes.

**Comment by jcfr** [comment]
Thanks @jamesobutler :100: and @lassoan :pray: 

Locally doing a test build, when it completes I will integrate.

**Pull request merged**